eva-server/src/main/java/uk/ac/ebi/eva/server/ws/VariantWSServer.java

```diff
@@ -47,40 +54,90 @@
     public QueryResponse getVariantById(@PathVariable("variantId") String variantId,
                                         @RequestParam(name = "studies", required = false) List<String> studies,
-                                        @RequestParam("species") String species,
+                                        @RequestParam(name = "species") String species,
+                                        @RequestParam(name = "annot-ct", required = false) List<String> consequenceType,
+                                        @RequestParam(name = "maf", required = false) String maf,
+                                        @RequestParam(name = "polyphen", required = false) String polyphenScore,
+                                        @RequestParam(name = "sift", required = false) String siftScore,
                                         HttpServletResponse response)
             throws IllegalOpenCGACredentialsException, UnknownHostException, IOException {
         initializeQueryOptions();
 
-        VariantDBAdaptor variantMongoDbAdaptor = DBAdaptorConnector.getVariantDBAdaptor(species);
 
-        if (studies != null && !studies.isEmpty()) {
-            queryOptions.put("studies", studies);
+        if (species.isEmpty()) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            return setQueryResponse("Please specify a species");
         }
 
-        if (!variantId.contains(":")) { // Query by accession id
-            return setQueryResponse(variantMongoDbAdaptor.getVariantById(variantId, queryOptions));
-        } else { // Query by chr:pos:ref:alt
-            String parts[] = variantId.split(":", -1);
-            if (parts.length < 3) {
-                response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
-                return setQueryResponse("Invalid position and alleles combination, please use chr:pos:ref or chr:pos:ref:alt");
-            }
+        MultiMongoDbFactory.setDatabaseNameForCurrentThread(DBAdaptorConnector.getDBName(species));
 
-            Region region = new Region(parts[0], Integer.parseInt(parts[1]), Integer.parseInt(parts[1]));
-            queryOptions.put("reference", parts[2]);
-            if (parts.length > 3) {
-                queryOptions.put("alternate", String.join(":", Arrays.copyOfRange(parts, 3, parts.length)));
-            }
+        List<VariantEntity> variantEntities;
+
+        if (variantId.contains(":")) {
+            variantEntities = queryByCoordAndAllele(variantId);
+        } else {
+            variantEntities = querybyId(variantId, studies, consequenceType, maf, polyphenScore, siftScore);
+        }
 
-            return setQueryResponse(variantMongoDbAdaptor.getAllVariantsByRegion(region, queryOptions));
+
+        QueryResult<VariantEntity> queryResult = new QueryResult<>();
+        queryResult.setResult(variantEntities);
+        return setQueryResponse(queryResult);
+    }
+
+    List<VariantEntity> queryByCoordAndAllele(String variantId) {
```

Comment:
Please rename to `queryByCoordinatesAndAlleles` and ensure the interfaces takes the arguments the method name declares: instead of taking a single string as argument, implement another auxiliary method to split the 3-4 parts of the string.